httpd24u.spec

```diff
@@ -66,32 +66,33 @@ Source52: htcacheclean.init
 Source53: httpd.logrotate-legacy
 # build/scripts patches
 Patch1: httpd-2.4.1-apctl.patch
-Patch2: httpd-2.4.9-apxs.patch
+Patch2: httpd-2.4.43-apxs.patch
 Patch3: httpd-2.4.1-deplibs.patch
 Patch6: httpd-2.4.3-apctl-systemd.patch
 Patch8: httpd-2.4.35-layout-legacy.patch
 # Needed for socket activation and mod_systemd patch
-Patch19: httpd-2.4.25-detect-systemd.patch
+# Patch19: httpd-2.4.25-detect-systemd.patch - now in upstream
 # Features/functional changes
 Patch23: httpd-2.4.33-export.patch
 Patch24: httpd-2.4.1-corelimit.patch
 Patch25: httpd-2.4.25-selinux.patch
 Patch26: httpd-2.4.4-r1337344+.patch
 Patch27: httpd-2.4.2-icons.patch
-Patch29: httpd-2.4.33-systemd.patch
+# Patch29: httpd-2.4.33-systemd.patch
 Patch30: httpd-2.4.4-cachehardmax.patch
 Patch34: httpd-2.4.17-socket-activation.patch
 
 # Bug fixes
 # https://bugzilla.redhat.com/show_bug.cgi?id=1397243
-Patch58: httpd-2.4.34-r1738878.patch
+# Patch58: httpd-2.4.34-r1738878.patch - now in upstream
 
 # Security fixes
 
 License: ASL 2.0
 Group: System Environment/Daemons
 BuildRequires: autoconf, perl, pkgconfig, findutils, xmlto
 BuildRequires: zlib-devel, libselinux-devel, lua-devel
+BuildRequires: epel-rpm-macros
```

Comment:
I needed this for my instance. It doesn't hurt your build system and will help those that are working on standalone systems for development.

Reply:
By that logic we should add BuildRequires for rpm-build, make, patch, redhat-rpm-config, etc.  That's not reasonable.

If your build system is using mock, add `config_opts['chroot_setup_cmd'] = 'install @buildsys-build'` to the config, similar to `/etc/mock/templates/epel-7.tpl`.  If you're using something else, just install `@buildsys-build` at the same time you install rpm-build and other toolchain packages.

Reply:
I have a CentOS 7 system that is configured per RHEL instructions to build RPMs. If something is in those instructions then it does not need to be included (rpm-build, make, patch, redhat-rpm-config, etc). epel-rpm-macros are not included in those. As I said before, this does nothing to hurt your environment and it makes sure that I have what I need in mine.

Reply:
@SteveSimpson, there is no guarantee that the macros provided by the `epel-rpm-macros` package will always remain in that package from anybody. The defined interface from the Fedora EPEL project is that your build environment has the EPEL repo activated and you've installed the `@buildsys-build` group. It seems like our documentation is slightly unclear here: IUS packages *must* be built using the Mock configuration files that are defined in our documentation.

As IUS requires EPEL at both build-time and run-time, there's basically no way the necessary macros wouldn't get installed. It's also important to note that `epel-rpm-macros` does not exist in Fedora, the ultimate upstream project where all IUS packages come from. The build dependency on `epel-rpm-macros` breaks building IUS packages with Fedora, which is fundamentally supposed to work, as we are backporting from Fedora to RHEL/CentOS.

SInce `epel-rpm-macros` is a shim package (containing backports from Fedora) that is already automatically installed by Mock accordingly, we require it to be present in the build environment, but ***not*** listed as a build dependency in spec files.

Reply:
Resolved in 39e6f26347c96e0b22197dfbf7501eec0103ec09.